WcaOnRails/app/views/competitions/_nearby_competitions.html.erb

```diff
@@ -11,6 +11,9 @@
   <div class="col-sm-9">
     <% if @competition.has_date? && @competition.has_location? %>
       <% nearby_competitions = @competition.nearby_competitions(Competition::NEARBY_DAYS_WARNING, Competition::NEARBY_DISTANCE_KM_WARNING)[0, 10] %>
+      <% if !current_user.can_view_hidden_competitions? %>
```

Comment:
Nice.

You _could_ build an `@alert_data = [:danger, '.no_date_yet']` etc, but that would get weirder than it's worth here.

Personally, I'd one line some of these, like `<%= alert :danger, t('.no_date_yet') unless @competition.has_date? %>`, but I know my preferences are a little extreme in that area, so only change it if you actually agree.